src/main/cpp/src/cast_string.cu

```diff
@@ -0,0 +1,204 @@
+/*
+ * Copyright (c) 2022, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "cast_string.hpp"
+
+#include <rmm/device_scalar.hpp>
+#include <rmm/exec_policy.hpp>
+
+#include <cudf/column/column.hpp>
+#include <cudf/column/column_device_view.cuh>
+#include <cudf/detail/utilities/cuda.cuh>
+#include <cudf/detail/utilities/integer_utils.hpp>
+#include <cudf/null_mask.hpp>
+
+#include <cub/warp/warp_reduce.cuh>
+
+#include <cooperative_groups.h>
+
+using namespace cudf;
+
+namespace spark_rapids_jni {
+
+namespace detail {
+
+constexpr auto NUM_THREADS{256};
+
+constexpr bool is_whitespace(char const chr)
+{
+  switch (chr) {
+    case ' ':
+    case '\r':
+    case '\t':
+    case '\n': return true;
+    default: return false;
+  }
+}
+
+template <typename T>
+__global__ void string_to_integer_kernel(T* out,
+                                         bitmask_type* validity,
+                                         const char* const chars,
+                                         offset_type const* offsets,
+                                         size_type num_rows)
+{
+  auto const group = cooperative_groups::this_thread_block();
+  auto const warp  = cooperative_groups::tiled_partition<cudf::detail::warp_size>(group);
+
+  // each thread takes a row and marches it and builds the integer for that row
+  auto const row = blockIdx.x * blockDim.x + threadIdx.x;
+  if (row >= num_rows) { return; }
+  auto const active        = cooperative_groups::coalesced_threads();
+  auto const row_start     = offsets[row];
+  auto const len           = offsets[row + 1] - row_start;
+  bool valid               = len > 0;
+  T thread_val             = 0;
+  int i                    = 0;
+  T sign                   = 1;
+  constexpr bool is_signed = std::is_signed_v<T>;
+
+  if (valid) {
+    // skip leading whitespace
+    while (is_whitespace(chars[i + row_start])) {
+      i++;
+    }
+
+    // check for leading +-
+    if constexpr (is_signed) {
+      if (chars[i + row_start] == '+' || chars[i + row_start] == '-') {
+        if (chars[i + row_start] == '-') { sign = -1; }
+        i++;
+      }
+    }
+
+    bool truncating = false;
+    for (int c = i; c < len; ++c) {
+      auto const chr = chars[c + row_start];
+      if (chr == '.') {
```

Comment:
It doesn't look like this handles multiple `.` characters appearing in the string properly.  Seems like it needs to be:
```suggestion
      if (!truncating && chr == '.') {
```

Reply:
Thanks, nice catch!